### PR TITLE
Initial PR

### PR DESCRIPTION
**Pull request by xoss**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,31 @@
...
+| Files / Volumes | Description |
+|---|---|
+| /var/run/docker.sock | Should be overwritten with the docker socket from host. |
+
+| ENV var | Description |
+|---|---|
+| DOCKER_SOCKET | Path to docker socket in container, defaults to `/var/run/docker.sock`. |
+| CRON_PATTERN | Pattern to run the cron, defaults to `0 */10 8-17 * * 1-5`. |
+| TERMINATE_WHITELIST | List of container names to match for termination. |
+| TERMINATE_BLACKLIST | List of container names to exlude for matching. |
```
Comment (xoss) [inline README.md]: exclude (!)

Comment (xoss) [inline README.md]: Nevertheless, I'd prefer sth. like:
`exclude from matching` or `exclude from termination`

Comment (bkonetzny) [inline README.md]: Fixed :)

File: README.md [hunk 2]
```diff
@@ -1 +1,31 @@
...
+| Files / Volumes | Description |
+|---|---|
+| /var/run/docker.sock | Should be overwritten with the docker socket from host. |
+
+| ENV var | Description |
+|---|---|
+| DOCKER_SOCKET | Path to docker socket in container, defaults to `/var/run/docker.sock`. |
+| CRON_PATTERN | Pattern to run the cron, defaults to `0 */10 8-17 * * 1-5`. |
+| TERMINATE_WHITELIST | List of container names to match for termination. |
+| TERMINATE_BLACKLIST | List of container names to exclude for matching. |
```
Comment (xoss) [inline README.md]: You were too fast to read my second comment.
It must be `exclude from matching` or `exclude from termination`

File: server.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
+var _ = require('lodash');
+var CronJob = require('cron').CronJob;
+var Docker = require('dockerode');
+
+var dockerSocketPath = process.env.DOCKER_SOCKET || '/var/run/docker.sock';
+var docker = new Docker({socketPath: dockerSocketPath});
+var cronPattern = process.env.CRON_PATTERN || '0 */10 8-17 * * 1-5';
+var maxKillCount = process.env.MAX_KILL_COUNT && parseInt(process.env.MAX_KILL_COUNT, 10) || 1;
+var randomLimit = 7;
```
Comment (xoss) [inline server.js]: It's rather unusual to define a probability at a scale of 0-10?
Why not `randomLimit = .7`?
Then, you don't need to multiple by factor `10` further down.

Comment (xoss) [inline server.js]: Why is this not configurable?

Comment (bkonetzny) [inline server.js]: Made configurable and defaulted to `.7`.

File: README.md (hunk 2 above)
Comment (bkonetzny) [inline README.md]: Fixed.

File: server.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+new CronJob(cronPattern, function(){
+  // Get random integer between 1 and allowed kill count.
+  var randomKillCount = _.random(1, maxKillCount);
+
+  console.log('[t800] Looking for %s of %s targets.', randomKillCount, maxKillCount);
+
+  var whitelist = null;
```
Comment (xoss) [inline server.js]: Why a `whitelist` and a `blacklist` after all?
It should be efficient enough to define a `whitelist`.
If we decide to support both, the blacklist should take precedence. It's easy to miss one entry in the whitelist, but adding that entry to the blacklist still does kill the container. Shouldn't we be conservative about that?

Comment (bkonetzny) [inline server.js]: Well it's `whitelist` OR `blacklist`. We either kill only service allowed in the whitelist (which needs to be adjusted every time we get a new container) OR we kill everything except the service listed in the blacklist (which is the preferred way, as every new container automatically get's available for killing).

File: server.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+  if (randomChaos < randomLimit) {
+    console.log('[t800] Randomly decided not to kill anyone: %s < %s', randomChaos, randomLimit);
+    return;
+  }
+
+  // Get all running docker containers from host.
+  docker.listContainers(function (err, containers) {
+    var targets = [];
+
+    containers.forEach(function (containerInfo) {
```
Comment (xoss) [inline server.js]: I'd prefer
```js
var targets = containers.filter(function (containerInfo) {
   return whitelist.indexOf(containerInfo.Names[0]) > 0 && blacklist.indexOf(containerInfo.Names[0]) === -1;
});

Comment (xoss) [inline server.js]: Still, don't you think this is cleaner, and faster as well.

```js
var targets = containers.filter(function (containerInfo) {
    var whitelisted = !!_.find(whitelist || [], function (lookup) {
      return containerInfo.Names[0].indexOf(lookup) > -1;
    });
    var blacklisted = !_.find(blacklist, function (lookup) {
      return containerInfo.Names[0].indexOf(lookup) > -1;
    });

  return whitelisted || !blacklisted;
});

Comment (bkonetzny) [inline server.js]: Updated logic to your approach, added some logging too.

**Pull request closed without merging**